core/deployment/src/main/java/io/quarkus/deployment/configuration/matching/PatternMapBuilder.java

```diff
@@ -19,18 +19,19 @@ private PatternMapBuilder() {
         for (RootDefinition rootDefinition : rootDefinitions) {
             final String rootName = rootDefinition.getRootName();
             ConfigPatternMap<Container> addTo = patternMap, child;
-            assert !rootName.isEmpty();
-            NameIterator ni = new NameIterator(rootName);
-            assert ni.hasNext();
-            do {
-                final String seg = ni.getNextSegment();
-                child = addTo.getChild(seg);
-                ni.next();
-                if (child == null) {
-                    addTo.addChild(seg, child = new ConfigPatternMap<>());
-                }
-                addTo = child;
-            } while (ni.hasNext());
+            if (!rootName.isEmpty()) {
+                NameIterator ni = new NameIterator(rootName);
+                assert ni.hasNext();
```

Comment:
The `assert` keyword is only triggered when `-ea` is specified in the `java` cmd line. Not sure it is a good idea to use it

Reply:
Perhaps changing the `do...while` loop with a `while` should be enough?

Reply:
I use assertions as a code comment to show conditions that are always true, and can be relied on without explicit testing.  This assertion has no side-effects.